### PR TITLE
Add documentation for publishing-analytics-events-to-external-systems

### PR DESCRIPTION
**Pull request by Lasith650**
This will add documentation for publishing analytics events to external systems.

And this documentation will cover below sections,

1. How to create a sample to log and publish analytics events.
2. How to configure the created sample with API-M 4.0.0.
3. How to publish logs to external systems (ELK).

### CODE REVIEW

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+---
+title: Publishing Analytics Events to External Systems - API Manager Documentation 4.0.0
+---
+
+# Publishing Analytics Events to External Systems
+
+## Introduction
+
+It is possible to log and publish analytics events to external systems. This will guide you on how to create a sample to log analytics events and to publish those logged events to an external system. For the demonstration purpose we have selected ELK as the external system.
```
Comment (fazlan-nazeem) [inline en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md:9]:
```suggestion
Instead of publishing analytics events to the cloud, It is also possible to log the same events and publish them to external systems. This guide will explain the steps required to do it. For demonstration purposes, we have selected ELK as the external system.
```

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    loggers = reporter, ...(list of other available loggers)
+
+Add bellow configurations after the loggers,
+
+    logger.reporter.name = <PackageName>
+    logger.reporter.level = INFO
+
+## Visualizing Logs
+
+WSO2 API-M logs are structured in a way that we can easily plug them in to a commercially available tool in order to visualize them. As a result this gives the capability to monitor analytics traffic for a particular API. For this purpose ELK is selected.
```
Comment (fazlan-nazeem) [inline en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md:181]:
```suggestion
WSO2 API-M logs are structured in a way that we can easily plug them into a log visualization available tool in order to visualize them. As a result, this gives the capability to monitor analytics traffic for a particular API. For this purpose, ELK is selected.
```

**Comment by fazlan-nazeem** [comment]
Where in this guide have you pointed to the GitHub sample? If you haven't, let's point it and say that a sample can be found in that location.

**Review by fazlan-nazeem — APPROVED** [review]

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md (hunk 1 above)
Comment (Lasith650) [reply en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md]: Fixed with ea12a8a83344ae33c5b9070693ff903eccd85600

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+    filebeat.inputs:
+    - type: log
+    enabled: true
+    paths:
+        - /<API-M HOME>/repository/logs/wso2carbon.log
+
+[Set up](https://www.elastic.co/guide/en/beats/filebeat/7.13/filebeat-installation-configuration.html#setup-assets) assets
+
+In case of a failure with the above command use below to set up assets,
```
Comment (fazlan-nazeem) [inline en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md:219]:
```suggestion
In case of a failure with the above command, use below to set up assets,
```

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md (hunk 2 above)
Comment (Lasith650) [reply en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md]: Fixed with ea12a8a83344ae33c5b9070693ff903eccd85600

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+[Start](https://www.elastic.co/guide/en/beats/filebeat/7.13/filebeat-installation-configuration.html#start) Filebeat
+
+### View Data on Kibana
+
+Filebeat comes with pre-built Kibana dashboards and UIs for visualizing log data.
+
+[Launch Kibana and discover](https://www.elastic.co/guide/en/beats/filebeat/7.13/filebeat-installation-configuration.html#view-data) log data.
+
+Once you have followed and completed above steps successfully you will be able to visualize log data as shown below,
```
Comment (fazlan-nazeem) [inline en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md:231]:
```suggestion
Once you have followed and completed the above steps successfully, you will be able to visualize log data as shown below,
```

File: en/mkdocs.yml [hunk 5]
```diff
@@ -2054,6 +2055,7 @@ plugins:
               'api-analytics/viewing/analytics-pages-latency.md': 'https://apim.docs.wso2.com/en/4.0.0/api-analytics/viewing/view-api-analytics-on-latency/'
               'api-analytics/viewing/analytics-pages-devices.md': 'https://apim.docs.wso2.com/en/4.0.0/api-analytics/viewing/view-analytics-on-devices/'
               'api-analytics/viewing/analytics-pages-report.md': 'https://apim.docs.wso2.com/en/4.0.0/api-analytics/viewing/download-api-analytics-reports/'
+              'api-analytics/samples/publishing-analytics-events-to-external-systems.md': 'https://apim.docs.wso2.com/en/4.0.0/api-analytics/samples/publishing-analytics-events-to-external-systems/'
```
Comment (Mariangela) [inline en/mkdocs.yml:2058]: Why did you add a redirection here? This is only needed if the page name or path changes

File: en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+![Logs listed in kibana]({{base_path}}/assets/img/analytics/samples/logs_listed_in_kibana.png)
+
+#### Filtering Total Analytics Traffic
+
+It is possible to view the analytics traffic by applying a filter as shown below,
+
+Replace `<MetricReporterImplClass>` with the class name given to the MetricReporter implementation class that you have created with your sample.
+
+![Total analytics traffic filter]({{base_path}}/assets/img/analytics/samples/total_analytics_traffic_filter.png)
```
Comment (Mariangela) [inline en/docs/api-analytics/samples/publishing-analytics-events-to-external-systems.md:241]: All images should be zoomable.
Do not add _ in image names. Instead replace it with - a hyphen.
